src/components/Pressable/Pressable.tsx

```diff
@@ -0,0 +1,247 @@
+          pressableRef.current?.measure((_x, _y, width, height) => {
+            // pressableRef.current?._nativeTag yields targetId on android & ios
+            if (
+              !isTouchWithinInset(
+                {
+                  width,
+                  height,
+                },
+                normalizedHitSlop,
+                event.changedTouches.at(-1)
+              ) ||
+              isPressedDown.current
+            ) {
+              return;
+            }
+
+            isPressedDown.current = true;
+
+            if (props.unstable_pressDelay) {
+              pressDelayTimeoutRef.current = setTimeout(() => {
+                pressInHandler(event);
+              }, props.unstable_pressDelay);
+            } else {
+              pressInHandler(event);
+            }
+          });
+        })
+        .onTouchesUp((event) => {
+          if (
+            !isPressedDown.current ||
+            event.allTouches.length > event.changedTouches.length
+          ) {
+            return;
+          }
+
+          if (
+            props.unstable_pressDelay &&
+            pressDelayTimeoutRef.current !== null
+          ) {
+            // if pressDelay is set, we want to call onPressIn on touch up
+            clearTimeout(pressDelayTimeoutRef.current);
+            pressInHandler(event);
+          }
+
+          props.onPressOut?.(adaptTouchEvent(event));
+
+          if (isPressCallbackEnabled.current) {
+            props.onPress?.(adaptTouchEvent(event));
+          }
+
+          isPressedDown.current = false;
+          setPressedState(false);
+        })
+        .onTouchesCancelled((event) => {
+          if (
+            !isPressedDown.current ||
+            event.allTouches.length > event.changedTouches.length
+          ) {
+            return;
+          }
+
+          // original triggers onPressOut on cancel, but not onPress
+          props.onPressOut?.(adaptTouchEvent(event));
+
+          isPressedDown.current = false;
+          setPressedState(false);
+        }),
+    [
+      props.onPress,
+      props.onPressIn,
+      props.onPressOut,
+      setPressedState,
+      isPressedDown,
+      isPressCallbackEnabled,
+      normalizedHitSlop,
+      pressDelayTimeoutRef,
+    ]
+  );
+
+  // rippleGesture lives inside RNButton to enable android's ripple
+  const rippleGesture = useMemo(() => Gesture.Native(), []);
+
+  pressGesture.minDuration(
+    (props.delayLongPress ?? DEFAULT_LONG_PRESS_DURATION) +
+      (props.unstable_pressDelay ?? 0)
+  );
+
+  const appliedHitSlop = addInsets(
+    normalizedHitSlop,
+    normalizedPressRetentionOffset
+  );
+
+  const isPressableEnabled = props.disabled !== true;
+
+  touchGesture.enabled(isPressableEnabled);
+  pressGesture.enabled(isPressableEnabled);
+  hoverGesture.enabled(isPressableEnabled);
+
+  touchGesture.shouldCancelWhenOutside(true);
+  pressGesture.shouldCancelWhenOutside(true);
+  hoverGesture.shouldCancelWhenOutside(true);
```

Comment:
Should it be set to `true`?

Reply:
Yes, setting it to `false` wouldn't cancel the gestures when they leave `pressRetentionOffset` area, which is how the legacy pressable behaves.
And there isn't a prop to set it to `false`.